app/javascript/components/auth-key-pair-cloud/auth-key-pair-cloud-form.schema.js

```diff
@@ -0,0 +1,48 @@
+import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
+import { API } from '../../http_api';
+
+const providerOptions = () => API.get(`/auth_key_pair_cloud/ems_form_choices`).then(data => [
+  ...data.ems_choices.map(({ id, name }) => ({ value: id, label: name })),
+
```

Comment:
The `...` into an array is kinda redundant, so you can remove that. Also we should ask for this information from the API instead. Try to send an `API.options('/api/auth_key_pair_cloud')` request and check if the data needed is there. If not, let me know and I'll make it happen. Eventually you should destructure it from the response and just use it.

Reply:
^